phys/module_mp_p3.F

```diff
@@ -242,35 +280,49 @@ SUBROUTINE p3_init(lookup_file_dir,nCat)
  dnu(15) =  0.966
  dnu(16) =  0.966
 
-! calibration factors for ice deposition and sublimation
-!   These are adjustable ad hoc factors used to increase or decrease deposition and/or
-!   sublimation rates.  The representation of the ice capacitances are highly simplified
-!   and the appropriate values in the diffusional growth equation are uncertain.
- clbfact_dep = 1.
- clbfact_sub = 1.
-
 !------------------------------------------------------------------------------------------!
 ! read in ice microphysics table
 
+ procnum = 0
+#ifdef ECCCGEM
+ call rpn_comm_rank(RPN_COMM_GRID,procnum,istat)
+ itab = 0.
+ itabcoll = 0.D0
+ itabcolli1 = 0.D0
+ itabcolli2 = 0.D0
+#endif
+
+ IF_PROC0: if (procnum == 0) then
+
  print*
  print*, ' P3 microphysics: v',version_p3
- print*, '   P3_INIT (reading/creating look-up tables) ...'
+ print*, '   P3_INIT (reading/creating look-up tables [v',trim(version_intended_table_1), &
+         ', v',trim(version_intended_table_2),']) ...'
 
- open(unit=10,file=lookup_file_1, status='old')
+ open(unit=10, file=lookup_file_1, status='old', action='read')
 
  !-- check that table version is correct:
- read(10,*) dumstr,version_header_table1
- if (trim(version_p3) /= version_header_table1(1:5)) then
+ !   note:  to override and use a different lookup table, simply comment out
+ !   the 'return' below, and the 'stop' if using WRF
+ read(10,*) dumstr,version_header_table_1
+ if (trim(version_intended_table_1) /= trim(version_header_table_1)) then
     print*
-    print*, '***   WARNING in P3_INIT   ***'
-    print*, ' P3 package version:     ',trim(version_p3)
-    print*, ' Lookup table 1 version: ',trim(version_header_table1)
-    print*, '******************************'
+    print*, '***********   WARNING in P3_INIT   *************'
+    print*, ' Loading lookupTable_1: v',trim(version_header_table_1)
+    print*, ' P3 v',trim(version_p3),' is intended to use lookupTable_1: v',    &
+            trim(version_intended_table_1)
+!   print*, '               -- ABORTING -- '
+    print*, '************************************************'
     print*
-   !stop
+    global_status = STATUS_ERROR
+    if (trim(model) == 'WRF') then
+       print*,'Stopping in P3 init'
+       stop
```

Comment:
Since this code is used in several models, I understand the need to not introduce WRF-only-isms. However, a "stop" statement could cause some troubles. Any way to ifdef in a wrf_error_fatal call here?